From c72b11e3f864b254097bd9b13bd085042d728fc8 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:21 -0400
Subject: [tools] perf/trace: Beautify socket 'family' arg

Message-id: <1408462094-14194-93-git-send-email-jolsa@redhat.com>
Patchwork-id: 88070
O-Subject: [PATCH RHEL7.1 BZ1131394 092/325] perf trace: Beautify socket 'family' arg
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit e10bce815d5c0e7eb793231e4334839b2cd0d81f
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Wed Sep 4 10:27:41 2013 -0300
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index eef26bd..671df94 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -255,6 +255,16 @@ static DEFINE_STRARRAY(fcntl_cmds);
 static const char *sighow[] = { "BLOCK", "UNBLOCK", "SETMASK", };
 static DEFINE_STRARRAY(sighow);
 
+static const char *socket_families[] = {
+ "UNSPEC", "LOCAL", "INET", "AX25", "IPX", "APPLETALK", "NETROM",
+ "BRIDGE", "ATMPVC", "X25", "INET6", "ROSE", "DECnet", "NETBEUI",
+ "SECURITY", "KEY", "NETLINK", "PACKET", "ASH", "ECONET", "ATMSVC",
+ "RDS", "SNA", "IRDA", "PPPOX", "WANPIPE", "LLC", "IB", "CAN", "TIPC",
+ "BLUETOOTH", "IUCV", "RXRPC", "ISDN", "PHONET", "IEEE802154", "CAIF",
+ "ALG", "NFC", "VSOCK",
+};
+static DEFINE_STRARRAY(socket_families);
+
 static size_t syscall_arg__scnprintf_open_flags(char *bf, size_t size,
             struct syscall_arg *arg)
 {
@@ -429,7 +439,9 @@ static struct syscall_fmt {
  { .name     = "setitimer",  .errmsg = true,
    .arg_scnprintf = { [0] = SCA_STRARRAY, /* which */ },
    .arg_parm  = { [0] = &strarray__itimers, /* which */ }, },
- { .name     = "socket",     .errmsg = true, },
+ { .name     = "socket",     .errmsg = true,
+   .arg_scnprintf = { [0] = SCA_STRARRAY, /* family */ },
+   .arg_parm  = { [0] = &strarray__socket_families, /* family */ }, },
  { .name     = "stat",     .errmsg = true, .alias = "newstat", },
  { .name     = "tgkill",     .errmsg = true,
    .arg_scnprintf = { [2] = SCA_SIGNUM, /* sig */ }, },
-- 
1.7.1